.github/workflows/main_push_and_pull_request_workflow.yml

```diff
@@ -4,7 +4,6 @@ on:
   push:
     branches: [ 'main' ]
   pull_request:
-    branches: [ 'main' ]
```

Comment:
Adding this in a separate commit to allow other chained PRs to build fully instead of waiting for main to be the base branch.